tests/src/python/test_qgsproject.py

```diff
@@ -0,0 +1,113 @@
+# -*- coding: utf-8 -*-
+"""QGIS Unit tests for QgsProject.
+
+.. note:: This program is free software; you can redistribute it and/or modify
+it under the terms of the GNU General Public License as published by
+the Free Software Foundation; either version 2 of the License, or
+(at your option) any later version.
+"""
+__author__ = 'Sebastian Dietrich'
+__date__ = '19/11/2015'
+__copyright__ = 'Copyright 2015, The QGIS Project'
+# This will get replaced with a git SHA1 when you do a git archive
+__revision__ = '$Format:%H$'
+
+import qgis
+
+from qgis.core import QgsProject, QgsMessageLog
+
+from utilities import getQgisTestApp, TestCase, unittest
+
+QGISAPP, CANVAS, IFACE, PARENT = getQgisTestApp()
+
+
+class TestQgsProject(TestCase):
+
+    def __init__(self, methodName):
+        """Run once on class initialisation."""
+        unittest.TestCase.__init__(self, methodName)
+        self.messageCaught = False
+
+    def test_makeKeyTokens_(self):
+        # see http://www.w3.org/TR/REC-xml/#d0e804 for a list of valid characters
+
+        invalidTokens = []
+        validTokens = []
+
+        # all test tokens will be generated by prepending or inserting characters to this token
+        validBase = u"valid";
+
+        # some invalid characters, not allowed anywhere in a token
+        # note that '/' must not be added here because it is taken as a separator by makeKeyTokens_()
+        invalidChars = u"+*,;<>|!$%()=?#\x01";
+
+        # generate the characters that are allowed at the start of a token (and at every other position)
+        validStartChars = u":_";
+        charRanges = [
+            (ord(u'a'), ord(u'z')),
+            (ord(u'A'), ord(u'Z')),
+            (0x00F8, 0x02FF),
+            (0x0370, 0x037D),
+            (0x037F, 0x1FFF),
+            (0x200C, 0x200D),
+            (0x2070, 0x218F),
+            (0x2C00, 0x2FEF),
+            (0x3001, 0xD7FF),
+            (0xF900, 0xFDCF),
+            (0xFDF0, 0xFFFD),
+            #(0x10000, 0xEFFFF),   while actually valid, these are not yet accepted by makeKeyTokens_()
```

Comment:
I'd say not. The characters ```0x10000 ... 0xEFFFF``` are outside of the [Basic Multilingual Plane](https://en.wikipedia.org/wiki/Plane_%28Unicode%29#Basic_Multilingual_Plane) and you need a [wide build Python](https://www.python.org/dev/peps/pep-0261/) which is not installed in the current Travic CI setup (at least for the Mac):
* [failed build on Mac](https://travis-ci.org/SebDieBln/QGIS/jobs/92021412#L3065)
* [the build on Linux](https://travis-ci.org/SebDieBln/QGIS/jobs/92021411) originally timed out and I can not rebuild because I already deleted the branch

And reading @olivierdalang's [comment on this](https://github.com/olivierdalang/QGIS/blob/0132a3e173e96e733db970d45ee966a1d2ff0bdb/src/core/qgsproject.cpp#L76) I think similar issues arise with the C++-files misinterpreting these characters.

The worst thing we will do is warn the plugin developer where there probably(!) is no reason to warn. Did anyone actually test a project file containing _wide unicode_ characters?